### ISSUE TITLE
injection "Symbol()" not found

### ISSUE THREAD

**Issue by tkddnr924** [smastrom/vue-global-loader#2]
# Version
Node: 20.5.0
Nuxt: ^3.8.1
Vue: ^3.3.8

# issue
<img width="1679" alt="스크린샷 2024-02-07 오전 10 52 51" src="https://github.com/smastrom/vue-global-loader/assets/15306531/72f703bb-659b-4bc9-9e99-c11f8446ccc3">

## issue text
```
[Vue warn]: injection "Symbol()" not found. 
  at <GlobalLoaderImpl __attrs= Object > 
  at <GlobalLoaderClientOnly> 
  at <GlobalLoader> 
  at <LoadingBar> 

[Vue warn]: Unhandled error during execution of setup function 
  at <GlobalLoaderImpl __attrs= Object > 
  at <GlobalLoaderClientOnly> 
  at <GlobalLoader> 
  at <LoadingBar> 

[Vue warn]: Unhandled error during execution of scheduler flush. This is likely a Vue internals bug. Please open an issue at https://new-issue.vuejs.org/?repo=vuejs/core 
  at <GlobalLoaderImpl __attrs= Object > 
  at <GlobalLoaderClientOnly> 
  at <GlobalLoader> 
  at <LoadingBar> 

Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'options')
    at useGlobalLoader (index.js?v=599c3edf:104:29)
    at setup (GlobalLoaderImpl.vue:34:68)
    at callWithErrorHandling (chunk-R7VYKG7T.js?v=599c3edf:1739:18)
    at setupStatefulComponent (chunk-R7VYKG7T.js?v=599c3edf:7800:25)
    at setupComponent (chunk-R7VYKG7T.js?v=599c3edf:7761:36)
    at mountComponent (chunk-R7VYKG7T.js?v=599c3edf:6395:7)
    at processComponent (chunk-R7VYKG7T.js?v=599c3edf:6361:9)
    at patch (chunk-R7VYKG7T.js?v=599c3edf:5834:11)
    at mountChildren (chunk-R7VYKG7T.js?v=599c3edf:6078:7)
    at processFragment (chunk-R7VYKG7T.js?v=599c3edf:6295:7)
```

This error occurred, is there a way to resolve it?

**Comment by smastrom** [comment]
Hi @tkddnr924, can you please show me your `nuxt.config.ts` and `app.vue files`?

**Comment by smastrom** [comment]
Hi @tkddnr924, this suggests that the `vue-global-loader/nuxt` module has not been installed.

Can you confirm the module has been added correctly to your `nuxt.config.ts`?

```
export default defineNuxtConfig({
   modules: ['vue-global-loader/nuxt'],
})
```

**Comment by smastrom** [comment]
Hi @tkddnr924, this suggests that the `vue-global-loader/nuxt` module has not been installed.

Can you confirm the module has been added correctly to your `nuxt.config.ts`?

```ts
export default defineNuxtConfig({
   modules: ['vue-global-loader/nuxt'],
})
```

**Comment by tkddnr924** [comment]
Thank you for your response.

I applied everything you wrote in Readme. (nuxt.config.ts...)

The first time you run it, it runs without any problems. However, sometimes that kind of error occurs, and the problem occurs throughout the nuxt.

I will also debug the issue as there may be conflicts with other codes in my code.

**Comment by BayBreezy** [comment]
Having the same issue @smastrom 
Here is a link to a reproduction https://stackblitz.com/edit/nuxt-starter-stb3dm?file=pages%2Findex.vue

**Comment by BayBreezy** [comment]
So when I passed the ` addPlugin: true` to the `globalLoader` object inside the `nuxt.config` file, the error went away.

@tkddnr924 , you have to set that option to `true` in your nuxt config file. It even fixed the issue in the repro above

**Comment by smastrom** [comment]
Thank you @BayBreezy for the repro. Unfortunately the issue is not related to the `addPlugin` option. Try to download the `.zip` from Stackblit and run `pnpm i && pnpm dev` and you will notice that the error persists once you call `useGlobalLoader` in any setup function.

This drove me crazy for one hour but then I eventually kinda undersood what's happening.

`vue-global-loader` is a _client-only_ plugin. Once on the client, the plugin is installed in the Vue app. The plugin provides the store globally which is then consumed by `useGlobalLoader`, `GlobalLoader` and any of your components/composables via `inject`.

Given that the error happens on the server (printed in that beautiful fullscreen error page in green) and that [I made sure](https://github.com/smastrom/vue-global-loader/blob/317716f3d0da1825b761d3aacc338a4ce8f4e39a/packages/vue-global-loader/core/store.ts#L71C1-L82C5) to return a "noop" store on the server to exactly void those inject errors (as the "real" store is only provided on the client):

```ts
export function useGlobalLoader(scopedOptions: Partial<GlobalLoaderOptions> = {}) {
   if (isSSR) {
      return {
         displayLoader: () => Promise.resolve(),
         destroyLoader: noop,
         updateOptions: noop,
         __onDestroyed: noop,
         __onDisplayed: noop,
         options: readonly(DEFAULT_OPTIONS),
         isLoading: readonly(ref(false)),
      }
   }

   const store = inject(injectionKey)
   // ...
}
```

I am almost certain this has nothing to do with this package. I think this is a recent Nuxt bug that happens in some edge cases. I'm personally using this package on different Nuxt < 3.11 apps and I never faced any issue, same for the demo links available in the readme. Maybe it has something to do with how the components are shipped (in this package I don't compile the components, instead I directly ship the `.vue` files).

At the end of the day why is Nuxt even trying to inject on the client? It's clear it should not by looking the above code.

However, by adding an extra layer of safety to the `inject` call, I can make sure that the returned value is never `undefined`, even if Nuxt decides somehow to call it on the server:

```ts
const store = inject(injectionKey, noopStore)
```
 
Expect this fix to be shipped within the next hour and thank you very much as always!

**Comment by BayBreezy** [comment]
Ahh, I see.
Thank you for taking the time to fix this, appreciate it 👍🏽 .

I am gonna check the release to see if it works soon.

**Comment by smastrom** [comment]
Hi again @BayBreezy, please update to the just-released `v0.9.9` which now extends Vite config upon module installation to include `vue-global-loader` to `optimizeDeps`. This also fixes the original issue that @tkddnr924 was having on the client.

**Comment by smastrom** [comment]
Hi again @BayBreezy, please update to the just-released `v0.9.9` which now extends Vite config upon module installation to include `vue-global-loader` to `optimizeDeps`. This also fixes the original issue that @tkddnr924 was having on the client.

Have a good one and thank you again!

**Comment by smastrom** [comment]
Hi again @BayBreezy, please update to the just-released `v0.9.9` which now extends Vite config upon module installation to include `vue-global-loader` to `optimizeDeps`. This also fixes the original issue that @tkddnr924 was having on the client.

Thank you again and cheers!
